lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -107,8 +107,12 @@ def url_for(path)
       def each_resource(url, &block)
         return if url.nil?
         response = get_and_parse(url)
-        response["data"].each { |i| yield i }
-        each_resource(response["links"]["next"], &block)
+        paging = response.is_a?(Hash)
+
+        resources = paging ? response["data"] : response
+        resources.each { |i| yield i }
+
+        each_resource(response["links"]["next"], &block) if paging
```

Comment:
Minor but would fetch_path be safer here in case the links section doesn't exist?